src/coreclr/jit/importercalls.cpp

```diff
@@ -1310,14 +1310,16 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             const bool         mustImportEntryBlock = gtIsRecursiveCall(methHnd) || actualCall->IsInlineCandidate() ||
                                               actualCall->IsGuardedDevirtualizationCandidate();
 
+            BasicBlock* entryBb = opts.IsOSR() ? fgEntryBB : fgFirstBB;
+
             // Only schedule importation if we're not currently importing.
             //
-            if (mustImportEntryBlock && (compCurBB != fgEntryBB))
+            if ((opts.IsInstrumentedOptimized() || opts.IsOSR()) && mustImportEntryBlock && (compCurBB != entryBb))
```

Comment:
This is an OSR only issue, as full method compilation will naturally start importing with `fgFirstBB`.